modular_skyrat/modules/vore/code/datums/components/vore.dm

```diff
@@ -0,0 +1,157 @@
+/datum/component/vore
+	var/list/obj/vbelly/bellies = list()
+	var/mob/living/owner
+	var/char_vars
+	var/vore_toggles = list()
+	var/selected_belly = 1
+	var/vore_enabled = TRUE
+	var/current_slot
+
+/datum/component/vore/Initialize()
+	if (!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	owner = parent
+	if (ishuman(owner))
+		var/mob/living/carbon/human/character = owner
+		current_slot = character.character_slot
+	RegisterSignal(owner, COMSIG_PARENT_PREQDELETED, .proc/handle_delete)
+	update_bellies(TRUE)
+
+/datum/component/vore/Destroy()
+	UnregisterSignal(owner, COMSIG_PARENT_PREQDELETED)
+	handle_delete()
+	. = ..()
+
+/datum/component/vore/proc/update_current_slot()
+	if (!isnull(current_slot))
+		return current_slot
+	if (ishuman(owner))
+		var/mob/living/carbon/human/character = owner
+		current_slot = character.character_slot
+		if (!isnull(current_slot))
+			return current_slot
+	current_slot = owner.client?.prefs.default_slot
+	return current_slot
+
+/datum/component/vore/proc/handle_delete()
+	SIGNAL_HANDLER
+
+	owner.release_belly_contents()
+
+/datum/component/vore/proc/get_belly_contents(bellynum, ref=FALSE, living=FALSE, as_string=FALSE, ignored=null, full=FALSE)
+	if (bellynum < 1 || bellynum > bellies.len)
+		return
+	var/list/belly_contents = bellies[bellynum].get_belly_contents(ref, living, as_string, ignored, full)
+	return belly_contents
+
+/datum/component/vore/proc/update_bellies(set_ref=FALSE) //update everything
+	if (!owner.client?.prefs?.vr_prefs)
+		return
+	var/datum/vore_prefs/vore = owner.client.prefs.vr_prefs
+	vore_toggles = vore.vore_toggles
+	char_vars = vore.char_vars
+	selected_belly = vore.selected_belly
+	vore_enabled = vore.vore_enabled
+	for (var/bellynum in 1 to vore.bellies.len)
```

Comment:
please use length() here, unless I'm seeing an older preview or something and you've already implemented length() judging by the manually resolved comment?